### PR TITLE
SVG Spriting

### PR DESCRIPTION
**Pull request by peterbraden**
https://github.com/crowdhouse/crowdhouse-webapp/pull/139/files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,8 @@
     "gulp-plumber": "^1.0.0",
     "gulp-sass": "^2.0.1",
     "gulp-sourcemaps": "^1.5.2",
+    "gulp-svg-sprite": "^1.2.7",
+    "gulp-svgmin": "^1.2.0",
```
Comment (lejoe) [inline package.json]: Does look like you don't need this gulp-svgmin or?

File: package.json [hunk 2]
```diff
@@ -36,6 +38,7 @@
     "merge-stream": "^0.1.7",
     "rsyncwrapper": "^0.4.2",
     "run-sequence": "^1.1.0",
+    "vinyl": "^0.5.1",
```
Comment (lejoe) [inline package.json]: Don't see the need for vinyl either

File: src/styles/styles.scss [hunk 3]
```diff
@@ -4,6 +4,7 @@
     "0.settings",
     "1.typography",
     "2.color",
+    "sprites",
```
Comment (lejoe) [inline src/styles/styles.scss]: Yeah that's why I'm not a fan of numbering files or folders :)
But I can leave with this for now.

**Comment by lejoe** [comment]
Don't have time to look at the scss code logic but I guess this works as they use it on crowdhouse.
The rest looks fine for me in this first version.

File: src/styles/styles.scss (hunk 3 above)
Comment (peterbraden) [inline src/styles/styles.scss]: Yeah, I'm not a big fan either. What are the pros?

File: package.json (hunk 2 above)
Comment (peterbraden) [inline package.json]: Yup, I just copied this from crowdhouse, we don't need them.

**Comment by peterbraden** [comment]
Actually, now I'm using this, I'm against the 1-letter mixin name.

`i(foo)` is less clear than `icon(foo)` and it's 3 more characters.

Lets do that?

**Comment by meodai** [comment]
@peterbraden & @lejoe that would be a bonescss discussion then.

The numbering is there so the order of insertion is clear. Its important that they are inserted in a certain order. So helpers for example are always at the end and all the settings are actually set before they are used.

**Comment by peterbraden** [comment]
@meodai I would agree if I hadn't tried 'f' for font first :)
I think 1 letters are fine when they are paired with an attribute - ie 'c' is obviously color when it appears after 'background-color' etc. But when you are @including, a little more context is nice. Like you were saying when we were going through the grid classes - longer names don't cost much here.

Re the numbering - Ordering is definitely important, but surely that should be specified in the 'styles.scss' imports with commenting etc. For me, when I open a file, I know the content of what I'm after, but probably not the number, so every time I have to list the files to see it. Perhaps there's a fuzzy file open plugin for vim, but it doesn't seem to justify itself to me. What are your thoughts on being more explicit where we include, rather than in the filenames?

**Comment by meodai** [comment]
@peterbraden I used font first, but changed to typography, because there are attributes like margin as well. But I like the argument of saying, its ok to be a one letter thing when it follows an attribute. But for the sake of consistency I think i would like all of them to be full words then. @include icon(), @include typography('bla'), @include size('small'), @include color('white') etc...

for the numbering: Its not like I'm actively reading the number every time :D. I think its the positive feedback I get most for bonescss: Oh this numbering thing is nice, it makes it very clear in what order to read and include the files... We can change it for skeleton. But I still think that it makes it very clear how the CSS Chain is. And why, for example a helper is overwriting your other rules without having to open the actual style.scss file.

**Comment by lejoe** [comment]
Re: numbering
I'm fine with the bones numbering. I just believe we can get rid of it on the project files as you want to be more flexible there

**Comment by peterbraden** [comment]
@meodai you're the expert on scss, so I defer you you here :) I just wonder what happens when we need to add a file (as we have for spriting) - do we do 3.5.Sprites 3.6.Sprites-helpers 3.5.5.Sprites-include-before-helper. 999.this-file-needs-to-be-last. And all of a sudden we have a z-index scenario for filenaming :)

**Comment by meodai** [comment]
well lets do this in an other pull request then. We can discuss about the pro's and con's there right?

**Comment by peterbraden** [comment]
SGTM

**Pull request closed without merging**